### PR TITLE
Sd 111: Implemented POST leagues/ endpoint

### PR DESCRIPTION
**Pull request by ndara**

### CODE REVIEW

File: Service/src/routes/league.py [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
+   endDate = datetime.strptime(body['end'], DATE_FORMAT)
+   if startDate > endDate:
+      return make_response(jsonify(EndBeforeStart=errors['endBeforeStart']), 400)
+   
+   leagueDuration = (endDate - startDate).days
+   print(str(leagueDuration))
+   if leagueDuration > DAYS_IN_YEAR:
+      return make_response(jsonify(LeagueDurationTooLong=errors['leagueDurationTooLong']), 400)
+
+   startPos = body.get('startPos') or 10000
```
Comment (sshaul) [inline Service/src/routes/league.py]: Do we need this default value? I feel like the start position should always have a value given by the user if they're starting a league.

File: Service/src/routes/league.py [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
-      return make_response(json.dumps(err.messages), 400)
-
+   startDate = datetime.strptime(body['start'], DATE_FORMAT)
+   endDate = datetime.strptime(body['end'], DATE_FORMAT)
+   if startDate > endDate:
+      return make_response(jsonify(EndBeforeStart=errors['endBeforeStart']), 400)
+   
+   leagueDuration = (endDate - startDate).days
+   print(str(leagueDuration))
+   if leagueDuration > DAYS_IN_YEAR:
```
Comment (sshaul) [inline Service/src/routes/league.py:30]: Why do we need to limit it to a year? Also, can we have leagues that don't have an end date (for those personal portfolios for users to just play with)?

File: tests/routesTests/PostLeagueTests.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+      self.assertEquals(response.status_code, 400)
+      self.assertTrue('InvalidField' in responseData[0])
+      self.assertEquals(responseData[0]['InvalidField'], 'start date must be in MM-DD-YYYY format')
+
+
+   def test_post_league_invalidEnd(self):
+      body = {
+         "name": 'myLeague',
+         "start": "08-24-2019",
+         "end": "08/30/2019",
```
Comment (sshaul) [inline tests/routesTests/PostLeagueTests.py:185]: Not really necessary but would this still work if someone put an invalid date like 08-35-2019?

File: Service/src/routes/league.py (hunk 1 above)
Comment (ndara) [reply Service/src/routes/league.py]: I like having defaults whenever possible...if the user 'should always' give the start pos, then that's fine. Just means the default value won't be used. Im gonna get rid of this magic number tho haha.

File: Service/src/routes/league.py (hunk 2 above)
Comment (ndara) [reply Service/src/routes/league.py:30]: I just wanted to set a limit for the sake of setting a limit. I'm gonna keep this here for now and if we decide in a meeting we don't want any limit I'll remove it.

File: tests/routesTests/PostLeagueTests.py (hunk 3 above)
Comment (ndara) [reply tests/routesTests/PostLeagueTests.py:185]: Good test idea! Lemme add a test for it and fix it if it breaks.

**Pull request merged**